### PR TITLE
Update documentation for default CookieBuilder options

### LINKED ISSUE #41871: CookieBuilder SameSite XML documentation is incorrect
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

The XML documentation of [CookieBuilder SameSite](https://github.com/dotnet/aspnetcore/blob/main/src/Http/Http.Abstractions/src/CookieBuilder.cs#L51) is incorrect. As mention here [SameSite Documentation](https://docs.microsoft.com/en-us/aspnet/core/security/samesite?view=aspnetcore-6.0#api-usage-with-samesite) and in this [Issue](https://github.com/dotnet/aspnetcore/issues/8212) the default value of SameSite is not always Unspecified. The default value varies for the different components.

### Expected Behavior

The behavior that I would expect is to remove this sentence `The default value is <see cref="SameSiteMode.Unspecified"/>` from the XML documentation and add the official URL where it is explained better i.e. https://docs.microsoft.com/en-us/aspnet/core/security/samesite?view=aspnetcore-6.0#api-usage-with-samesite.

### Steps To Reproduce

_No response_

### Exceptions (if any)

_No response_

### .NET Version

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by sebastienros**
Fixes #41871

### CODE REVIEW

**Comment by Tratcher** [comment]
Others:
https://github.com/dotnet/aspnetcore/blob/8c1d2de742aa5ec37f086ab69eb446cf1abd1a22/src/Security/CookiePolicy/src/CookiePolicyOptions.cs#L35

What about https://github.com/dotnet/aspnetcore/issues/41871#issuecomment-1144198818?
CookieBuilder api docs could be modified to say "the default is X, but components utilizing CookieBuilder may use a different default."

File: src/Security/Authentication/Core/src/RemoteAuthenticationOptions.cs [hunk 1]
```diff
@@ -137,6 +137,18 @@ public new RemoteAuthenticationEvents Events
...
+    /// <remarks>
+    /// <para>
+    /// If an explicit <see cref="CookieBuilder.Name"/> is not provided, the system will automatically generate a
+    /// unique name that begins with <c>".AspNetCore.Correlation."</c>.
+    /// </para>
+    /// <para>
+    /// <see cref="CookieBuilder.SameSite"/> defaults to <see cref="SameSiteMode.None"/>.
+    /// <see cref="CookieBuilder.HttpOnly"/> defaults to <c>true</c>.
+    /// <see cref="CookieBuilder.IsEssential"/> defaults to <c>true</c>.
+    /// <see cref="CookieBuilder.SecurePolicy"/> defaults to <see cref="CookieSecurePolicy.SameAsRequest"/>.
```
Comment (JamesNK) [inline src/Security/Authentication/Core/src/RemoteAuthenticationOptions.cs:149]: Docs doesn't care about the line breaks. These will get squashed together into one paragraph.

Perhaps use a list? https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/xmldoc/recommended-tags#list

You can preview what XML docs look like by hovering mouse over API or using it from test to see IntelliSense

Comment (JamesNK) [reply src/Security/Authentication/Core/src/RemoteAuthenticationOptions.cs:149]: Applies to other lists in this PR.

File: src/Http/Http.Abstractions/src/CookieBuilder.cs [hunk 2]
```diff
@@ -41,7 +41,8 @@ public class CookieBuilder
     public virtual string? Domain { get; set; }
 
     /// <summary>
-    /// Indicates whether a cookie is inaccessible by client-side script.
+    /// Indicates whether a cookie is inaccessible by client-side script. The default value is <c>False</c>
+    /// but components utilizing may use a different value.
```
[the remark this answers is not in the document]
Comment (JamesNK) [reply src/Http/Http.Abstractions/src/CookieBuilder.cs:45]: Applies to other updates in this file.

File: src/Security/Authentication/Core/src/RemoteAuthenticationOptions.cs (hunk 1 above)
Comment (sebastienros) [reply src/Security/Authentication/Core/src/RemoteAuthenticationOptions.cs:149]: I agree, here is how it looks right now on the one I took for reference.
https://learn.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.antiforgery.antiforgeryoptions.cookie?view=aspnetcore-7.0#remarks

**Comment by sebastienros** [comment]
@Tratcher GTG?

There isn't any conceptual documentation to link to for CookieBuilder https://github.com/dotnet/AspNetCore.Docs/issues/5124

File: src/Security/Authentication/Cookies/src/CookieAuthenticationOptions.cs [hunk 3]
```diff
@@ -34,32 +34,20 @@ public CookieAuthenticationOptions()
...
-    /// <see cref="CookieBuilder.HttpOnly"/> determines if the browser should allow the cookie to be accessed by client-side javascript.
-    /// The default is true, which means the cookie will only be passed to http requests and is not made available to script on the page.
-    /// </para>
-    /// <para>
-    /// <see cref="CookieBuilder.Expiration"/> is currently ignored. Use <see cref="ExpireTimeSpan"/> to control lifetime of cookie authentication.
-    /// </para>
+    /// system uses the cookie authentication handler multiple times.</description></item>
+    /// <item><description><see cref="CookieBuilder.SameSite"/> determines if the browser should allow the cookie to be attached to same-site or cross-site requests.
+    /// The default is <c>Lax</c>, which means the cookie is only allowed to be attached to cross-site requests using safe HTTP methods and same-site requests.</description></item>
+    /// <item><description><see cref="CookieBuilder.HttpOnly"/> determines if the browser should allow the cookie to be accessed by client-side javascript.
```
Comment (JamesNK) [inline src/Security/Authentication/Cookies/src/CookieAuthenticationOptions.cs:46]:
```suggestion
    /// <item><description><see cref="CookieBuilder.HttpOnly"/> determines if the browser should allow the cookie to be accessed by client-side JavaScript.
```

File: src/Security/Authentication/Cookies/src/CookieAuthenticationOptions.cs [hunk 4]
```diff
@@ -34,32 +34,20 @@ public CookieAuthenticationOptions()
...
+    /// The default is <c>true</c>, which means the cookie will only be passed to http requests and is not made available to script on the page.</description></item>
```
Comment (JamesNK) [inline src/Security/Authentication/Cookies/src/CookieAuthenticationOptions.cs:47]:
```suggestion
    /// The default is <c>true</c>, which means the cookie will only be passed to HTTP requests and is not made available to JavaScript on the page.</description></item>
```

**Review by Tratcher — APPROVED** [review]

**Comment by captainsafia** [comment]
Triage: we're going to merge this. It's been reviewed and doc changes look good. 👍🏽

**Pull request merged**